node/src/components/storage.rs

```diff
@@ -473,8 +478,46 @@ impl Storage {
             fault_tolerance_fraction,
             max_ttl,
             metrics,
+            force_resync_pid_file: None,
         };
 
+        if force_resync {
```

Comment:
Can we avoid running "indexing block store", "initialize_block_body_db", etc. in case `force_resync` is set to `true`? (and if `persist_completed_blocks()` did not fail?)

Reply:
I don't think we can. What this setting is supposed to do is just clear the `completed_blocks` index, which will cause the node to sync again gathering any missing information, but I think we're supposed to keep the stuff we have to have it function properly and efficiently.

Reply:
So the `force_resync` is not intended to resync from scratch, but just to fill potentially missing parts?

Reply:
As far as I understand, an operator turns on `force_resync` to get any information about the state of the chain, such as additional finality signatures. It is not intended to wipe anything from the database except the `completed_blocks` index, which is just an indication of what blocks are considered "good enough" to not request additional information about them.

Reply:
If that's the case, the code as is makes perfect sense, thanks for explanation.

So it's just the parameter name that could be confusing, because it may suggest that we wipe the data and resync. But that's just a minor nit and we can leave it be.